File: python/twosum.py
# Input: nums = [2,7,11,15], target = 9
# Output: [0,1]
# Output: Because nums[0] + nums[1] == 9, we return [0, 1].

def two_sum(array, target):
    if len(array) > 1:
        for ele in array:
            remaining = target - ele
            if remaining in array:
                remaining_index = array.index(remaining)
                return [array.index(ele), remaining_index]
        return False
    else:
        return False
nums = [1, 2, 4, 6,  7, 8, 11, 15]
target = 9
print(two_sum(nums, target))

File: php/twoSum.php
<?php
/*
	Given an array of integers, return indices of the two numbers such that they add up to a specific target.
	You may assume that each input would have exactly one solution, and you may not use the same element twice.
	
	Given nums = [2, 7, 11, 15], target = 9,
	Because nums[0] + nums[1] = 2 + 7 = 9,
	return [0, 1].

*/
function twoSum($inputArray, $targetNumber){
	$outputIndex = array();
	for($i=0; $i<count($inputArray); $i++){
		$firstElement=$inputArray[$i];
		$secondElement=$targetNumber - $firstElement;
		if(array_search($secondElement, $inputArray)){
			if(!array_search("$secondElement,$firstElement", $outputIndex)){
				array_push($outputIndex,"$firstElement,$secondElement");
			}
		}
	}
	return $outputIndex;
}
$inputArray = [1, 2, 4, 6,  7, 8, 11, 15];
$targetNumber = 9;
var_export(twoSum($inputArray, $targetNumber));
?>
